### ISSUE TITLE
Shown event participants runs into a time-out for some events und users

### ISSUE THREAD

**Issue by Marko-ilmkreis** [hzi-braunschweig/SORMAS-Project#4088]
### Bug Description

I have reports from users that they have problems to open the list of certain event-participants. 

if user A open list the participants of event Z and 
if user B open list the participants of event Y they are running into a timeout after the waiting screen with sorting bubbles appears.

But all other users are able to open the list and thee participants. Even user B can see ths list of event Z.

The problem is still there even if we revoke and re-give the rights to the users, the cache of the browser is cleared and the browser is restarted.

It is possible to reproduce the error with the particular user.

For other events, the respective user can see all participants.

### System Details
* Device: Win10
* SORMAS version:  1.52.1
* Android version/Browser: Firefox/Edge
* User Role: Clinician, Contact Supervisor, Surveillance Supervisor

**Comment by DavidBaldsiefen** [comment]
Just for clarification, do you mean that generally no two users can access the participant list of any events at the same time? And it does not matter which events are opened?

**Comment by Marko-ilmkreis** [comment]
@DavidBaldsiefen No. From some times we have the problem that individual users (e.g. user A) cannot see the participants of exactly one event (e.g. event Z). The error can be reproduced at will with the respective user, even after days. In the meantime, all other users can see the participants of event Z.

We had the problem for the first time last Friday (15.01.) with a colleague and thought at first that the instance was suffering from low performance. We then asked Netzlink to increase the performance, which happened on Friday afternoon. The server was also restarted for this.

On Monday (18.01.), the colleague tried again to view the participants of this event, but this again resulted in a timeout.I could see the participants, as could a test user without admin rights. We put this down to coincidence until the next colleague contacted me yesterday, who also had the same problem with another event.

Both events have a larger number of participants (90 and 200).

**Comment by DavidBaldsiefen** [comment]
Ok I understand.
A general performance increase is already on the way for the event participant directory for the next release (1.55), but I am not sure if it will solve the issue. Do you get any error message with the timeout?

**Comment by MateStrysewske** [comment]
@DavidBaldsiefen This doesn't look like a performance problem to me though, because the list in question only contains three event participants according to the original post, and since other users can load it without any issues, it sounds like there's something else going wrong here.

**Comment by DavidBaldsiefen** [comment]
> 
> 
> @DavidBaldsiefen This doesn't look like a performance problem to me though, because the list in question only contains three event participants according to the original post, and since other users can load it without any issues, it sounds like there's something else going wrong here.

He just said both lists contain a larger number of participants in the last comment, or did I miss something? But yea I agree, it's unlikely that this is an performance issue. Still a weird issue though as the eventparticipants methods are not that different from all the other methods where IndexLists are retrieved, and I haven't heard that issue popping up anywhere else (did you?).

It might be very helpful to get access to the logfiles of the server in question.

**Comment by Marko-ilmkreis** [comment]
I talked to the colleague with first issue form Friday again. She tried to access the list yesterday again, with no success. Currently there are 52 participants. She told me also that they tried to access the list yesterday from different account, which ends in the time-out. 

> Do you get any error message with the timeout?

It just runs with the "sorting bubbles" and the progress bar on top of the window. I will ask them to try again tomorrow as we since today have deactivated MSers and Reports and I feel a increasing speed.

I come back tomorrow to this issue.

**Comment by DavidBaldsiefen** [comment]
@Marko-ilmkreis Any update on the issue? Does the problem still persist?

**Comment by Marko-ilmkreis** [comment]
@DavidBaldsiefen I do apologize for the late reply. First  was not in office and then the colleague in home office.
There was a misunderstanding between me and my colleague. The problem occurs when you try to list the contacts for an event. Then the contact list opens and the page usually shows the names greyed-out, but the "waiting balls" continue to run. If you resize the window (Firefox), the names are shortly visible before the waiting window appears again.

![Screenshot_2021-01-28 SORMAS](https://user-images.githubusercontent.com/74791118/106128786-faa77d00-615f-11eb-8871-fb2f0d827349.png)


I was able to reproduce this, also with another event (33 contacts). As in the test-Sormas are not that many contacts, I could not test it there.

The problem does not occur when 

1) The event has only a few contacts
2) when all event participants are displayed and 
3) when displaying cases of the event.

**Comment by DavidBaldsiefen** [comment]
I could not replicate this on the development branch of 1.55. The test event contained 30 participants, 4 cases and 57 contacts.

**Comment by Marko-ilmkreis** [comment]
@DavidBaldsiefen If I enter the event-UUID manually, the result is the same. It runs with the bubble with no chance to cancel it. I tested it with an event with 18 contacts right now again and it goes the same way.

The problem is, that we are currently still working with 1.52.1 and there might be changes in Sormas which solving the problem in 1.55.

**Comment by DavidBaldsiefen** [comment]
I tried it on 1.52.1 and still couldn't replicate. (10 Participants, 7 Cases, 35 Contacts)

Maybe it is some weird case/contact/participant constellation causing this, or some other database issue. How many contacts are in your database in total?

Maybe it would make sense to set up a call where we can discuss this problem in more detail.
